app/src/components/alert/Alert.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { ReactNode, SyntheticEvent } from "react";
+import { css } from "@emotion/react";
+
+import { classNames } from "@phoenix/components";
+
+import { Text } from "../content";
+import { CloseOutline, Icon } from "../icon";
+import { SeverityLevel } from "../types";
+
+import { getSeverityIcon } from "./getSeverityIcon";
+import { useSeverityStyle } from "./useSeverityStyle";
+export interface AlertProps {
+  variant: SeverityLevel;
+  children?: ReactNode;
+  /**
+   * Title of the alert. Optional
+   */
+  title?: ReactNode;
+  /**
+   * A custom icon to show
+   */
+  icon?: ReactNode;
+  /**
+   * Whether or not an icon is shown on the left
+   * @default true
+   */
+  showIcon?: boolean;
+  /**
+   * If set to true, a close button is rendered
+   * @default false
+   */
+  dismissable?: boolean;
+  /**
+   * dismiss callback
+   */
+  onDismissClick?: (e: SyntheticEvent<HTMLButtonElement>) => void;
+  /**
+   * If set to true, this alert is being placed at the top of a page
+   * @default false
+   */
+  banner?: boolean;
+  /**
+   * Extra content (typically a button) added to the alert
+   */
+  extra?: ReactNode;
+}
+
+export const Alert = ({
+  variant,
+  title,
+  icon,
+  children,
+  showIcon = true,
+  dismissable = false,
+  onDismissClick,
+  banner = false,
+  extra,
+  ...otherProps
+}: AlertProps) => {
+  const variantStyle = useSeverityStyle(variant);
+
+  if (!icon && showIcon) {
+    icon = getSeverityIcon(variant);
+  }
+  return (
+    <div
+      {...otherProps}
+      css={css`
+        padding: var(--ac-global-dimension-static-size-100)
+          var(--ac-global-dimension-static-size-200);
+        border-radius: 4px;
+        color: var(--ac-global-text-color-900);
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+        &.ac-alert--banner {
+          border-radius: 0;
+          border-left: 0px;
+          border-right: 0px;
+        }
+        ${variantStyle}
```

Comment:
I think it would be good to declare this outside of the component so that the generated css is a stable reference.

I am also wondering about all of the string interpolation going in here. Could we use data attributes on the element and selectors in the styles so that all of the conditional logic happens within css instead of on the JS main thread? Check out the other ported components like Button and Token to see how we use data attributes to style components, as well as the small explainer here https://react-spectrum.adobe.com/react-aria/styling.html#states